### PR TITLE
Feat: delete template button

### LINKED ISSUE #3688: Add support for deleting templates from the UI
Can a button be added to the workspace template webpage to support template admins deleting a workspace template from the Coder website? I understand this can be done today using the CLI tool (e.g. `coder templates delete`), but I think it would be great if template admins could delete templates from the Coder website too.

### PR DESCRIPTION
**Pull request by presleyp**
Closes #3688 

- extracts DropdownButton component from WorkspacePage to reuse between that page and TemplatePage
- moves ErrorSummary to under PageHeader in WorkspacePage for better style; same placement on TemplatePage
- made the gap between sections smaller in WorkspacePage to match TemplatePage
- deletion shows a success toast and redirects to TemplatesPage
- error shows dismissible ErrorSummary
- delete button only visible to those with permission

Ideas for later: 
- disable delete button if workspaces are using this template, add tooltip to explain that this is the blocker
- make dropdown buttons have the same width as the collapsed button

Screen Shot 2022-08-31 at 2 23 26 PM

### CODE REVIEW

File: site/src/components/DropdownButton/DropdownButton.tsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+}) => {
+  const styles = useStyles()
+  const anchorRef = useRef<HTMLButtonElement>(null)
+  const [isOpen, setIsOpen] = useState(false)
+  const id = isOpen ? "action-popover" : undefined
+
+  /**
+   * Ensures we close the popover before calling any action handler
+   */
+  useEffect(() => {
```
Comment (BrunoQuaresma) [inline site/src/components/DropdownButton/DropdownButton.tsx:30]: What is the use case for this?

File: site/src/pages/TemplatePage/TemplatePageView.tsx [hunk 2]
```diff
@@ -36,104 +39,141 @@ export interface TemplatePageViewProps {
...
+                      action: "delete",
+                      button: (
+                        <DeleteButton handleAction={() => handleDeleteTemplate(template.id)} />
+                      ),
+                    },
+                  ]}
+                  canCancel={false}
+                />
+              ) : (
+                createWorkspaceButton("")
```
Comment (BrunoQuaresma) [inline site/src/pages/TemplatePage/TemplatePageView.tsx:108]: We could change the arg to be optional, it is not intuitive what an empty string means here.

**Review by BrunoQuaresma — APPROVED** [review]
LGTM. Just minor comments/questions.

File: site/src/components/DropdownButton/DropdownButton.tsx (hunk 1 above)
Comment (presleyp) [reply site/src/components/DropdownButton/DropdownButton.tsx]: Good catch, I meant to come back to this. It used to take a dependency that's no longer available, but I don't think it's the best way to handle it anyway. I switched to closing the popover on blur.

**Pull request merged**